Testing/missing tests

stanislav-halyn opened the pull request.

stanislav-halyn committed:
added tests for `find-by-ids` endpoint

stanislav-halyn committed:
Merge remote-tracking branch 'origin/develop' into testing/missing-tests

stanislav-halyn committed:
added tests for `upload` endpoint

tiagojsag committed:
Add error logging to s3 file upload failure

tiagojsag committed:
Add missing S3 credentials mock to travis config file

stanislav-halyn committed:
added tests for `flush` endpoint

stanislav-halyn committed:
Merge branch 'testing/missing-tests' of https://github.com/resource-watch/dataset into testing/missing-tests

stanislav-halyn committed:
added tests for `flush` endpoint

stanislav-halyn committed:
added a testcase for `flush` enpoint

stanislav-halyn committed:
added test cases for `/verification` endpoint

tiagojsag commented on app/test/e2e/dataset-find-by-ids.spec.js:
```diff
@@ -0,0 +1,78 @@
+        response.status.should.equal(200);
+        response.body.should.have.property('data').with.lengthOf(1);
+
+        response.body.data[0].id.should.equal(cartoFakeDataset.id);
```
please test the response body properly. You don't need to cover every single field, but testing just the IDs is not very helpful to prevent, for example, a bug in a serializer

tiagojsag commented on app/test/e2e/dataset-find-by-ids.spec.js:
```diff
@@ -0,0 +1,78 @@
+    it('Return 404 error if no data provided at all', async () => {
+        const response = await requester.post(`${BASE_URL}/find-by-ids`).send();
+
+        response.status.should.equal(404);
```
validate the error message too

tiagojsag commented on app/test/e2e/dataset-flush.spec.js:
```diff
@@ -0,0 +1,77 @@
+            .field('loggedUser', JSON.stringify(ROLES.ADMIN))
+            .send();
+
+        responseManager.status.should.equal(200);
```
not sure if there's more content in the reply, but if there is, it should be covered by your tests

stanislav-halyn replied:
there is an empty body

tiagojsag commented on app/test/e2e/dataset-verification.spec.js:
```diff
@@ -0,0 +1,142 @@
+
+});
+
+const stamperyResponseObject = {
```
I'd suggest moving this to test.contants.js, to keep this file easier to read

stanislav-halyn committed:
- fix eslint issues
- add test body contents
- add test error messages
- add validation for enpoint /dataset/find-by-ids

tiagojsag committed:
Minor CS changes to tests
